code/fraction/logger/reallogger.py

```diff
@@ -1,11 +1,16 @@
 import logging
+import os
 
 from fraction.logger.ilogger import ILogger
 
 
 class RealLogger(ILogger):
     def __init__(self):
         super(RealLogger, self).__init__()
+        try:
+            os.makedirs('tmp')
```

Comment:
This looks quite suspicious... First of all, there is build-in `tempfile` module in Python which provides all the proper wrappers... Then, logs should not be in "tmp" folder of the main app, it should be either "logs" folder or in system Temp.
And IIRC `logging` should be smart enough and create folders when needed...

Reply:
Thank you. Now there is "log" folder instead "temp".
I wish `logging` could create the folder :(